src/Sulu/Bundle/AdminBundle/Controller/IconController.php

```diff
@@ -0,0 +1,171 @@
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Rest\ListBuilder\CollectionRepresentation;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class IconController extends AbstractRestController implements ClassResourceInterface
+{
+    use RequestParametersTrait;
+
+    /**
+     * @var array<string, string>
+     */
+    private array $iconSets;
+
+    private FilesystemAdapter $cache;
+
+    /**
+     * @param array<string, string> $iconSets
+     */
+    public function __construct(
+        ViewHandlerInterface $viewHandler,
+        array $iconSets,
+    ) {
+        parent::__construct($viewHandler);
+        $this->iconSets = $iconSets;
+        $this->cache = new FilesystemAdapter();
+    }
+
+    /**
+     * Returns icons.
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        /** @var string $iconSetName */
+        $iconSetName = $this->getRequestParameter($request, 'icon_set', true);
+        $iconSet = \explode('://', $this->iconSets[$iconSetName]);
+        $search = $request->query->get('search');
+        $provider = $iconSet[0];
+        $path = $iconSet[1] ?? '';
+
+        // Get icons depending on the provider type.
+        switch ($provider) {
+            case 'svg':
+                $icons = $this->getIconsAsSVGs($path);
+                break;
+            case 'icomoon':
+                $icons = $this->getIconsAsIcomoon($path);
+                break;
+            default:
+                throw new BadRequestException();
+        }
+
+        // Implement a simple search functionality.
+        if ($search) {
+            $filteredIcons = [];
+
+            foreach ($icons as $icon) {
+                if (\str_contains($icon['id'], $search)) {
+                    $filteredIcons[] = $icon;
+                }
+            }
+
+            $icons = $filteredIcons;
+        }
+
+        // Sort by ID.
+        \usort($icons, fn ($a, $b) => $a['id'] <=> $b['id']);
+
+        return $this->handleView(
+            $this->view(
+                new CollectionRepresentation(
+                    $icons,
+                    'icons'
+                )
+            )
+        );
+    }
+
+    /**
+     * Return a set of SVG icons from a specific directory.
+     *
+     * @return array<array{id: string, content: string}>
+     */
+    public function getIconsAsSVGs(string $path)
+    {
+        $cacheKey = 'icons_' . \md5($path);
+
+        return $this->cache->get($cacheKey, function() use ($path) {
+            $icons = [];
+
+            $finder = new Finder();
+            $finder->in($path);
+
+            foreach ($finder as $file) {
+                $icons[] = [
```

Comment:
can we check here the filetype and skip non svg files, otherwise this will most likely crash